helix-core/src/comment.rs

```diff
@@ -45,9 +45,11 @@ fn find_line_comment(
 
             // determine margin of 0 or 1 for uncommenting; if any comment token is not followed by a space,
             // a margin of 0 is used for all lines.
-            if matches!(line_slice.get_char(pos + token_len), Some(c) if c != ' ') {
-                margin = 0;
-            }
+            margin = match line_slice.get_char(pos + token_len) {
+                Some(c) if c != ' ' => 0,
+                None => 0,
+                _ => margin,
+            };
```

Comment:
I think keeping this as an if is slightly cleaner

```suggestion
            if !matches!(line_slice.get_char(pos + token_len), Some(c) if c == ' '){
                margin = 0
            }
```